src/main/java/com/wolox/training/models/Book.java

```diff
@@ -58,6 +55,46 @@
     @JsonBackReference
     private List<User> users = new ArrayList<>();
 
+    public void setAuthor(String author) {
+        String message = String.format(Constants.CHECK_NULL_MESSAGE, "author");
+        this.author = Preconditions.checkNotNull(author, message);
```

Comment:
I would also check the non-emptyness of the String arguments, adapt others as needed:
```suggestion
        this.author = Preconditions.checkArgument(String.isNullOrEmpty(author), message);
```